crates/proof-of-sql/src/proof_primitive/hyrax/base/hyrax_commitment_evaluation_proof.rs

```diff
@@ -0,0 +1,105 @@
+use super::{
+    hyrax_commitment::HyraxCommitment,
+    hyrax_configuration::HyraxConfiguration,
+    hyrax_helpers::{compute_dynamic_vecs, matrix_size, row_and_column_from_index},
+    hyrax_public_setup::HyraxPublicSetup,
+    hyrax_scalar::HyraxScalarWrapper,
+};
+use crate::base::{
+    commitment::CommitmentEvaluationProof,
+    if_rayon,
+    proof::{ProofError, Transcript},
+    scalar::Scalar,
+};
+use alloc::{vec, vec::Vec};
+#[cfg(feature = "rayon")]
+use rayon::iter::{
+    IndexedParallelIterator, IntoParallelIterator, IntoParallelRefIterator, ParallelIterator,
+};
+use serde::{Deserialize, Serialize};
+
+/// The Hyrax evaluation proof scheme
+#[derive(Serialize, Deserialize)]
+pub struct HyraxCommitmentEvaluationProof<C: HyraxConfiguration> {
+    /// Represents the matrix cross the high vector.
+    /// Verified against the product with the low vector.
+    /// Verified against the commitment and high vector using the generators.
+    pub witness: Vec<C::OperableScalar>,
+}
+
+impl<C: HyraxConfiguration> CommitmentEvaluationProof for HyraxCommitmentEvaluationProof<C>
+where
+    for<'b> C::OperableGroup: 'b,
+{
+    type Scalar = HyraxScalarWrapper<C::OperableScalar>;
+
+    type Commitment = HyraxCommitment<C>;
+
+    type Error = ProofError;
+    type ProverPublicSetup<'a> = HyraxPublicSetup<'a, C::OperableGroup>;
+    type VerifierPublicSetup<'a> = HyraxPublicSetup<'a, C::OperableGroup>;
+
+    fn new(
+        _transcript: &mut impl Transcript,
+        a: &[Self::Scalar],
+        b_point: &[Self::Scalar],
+        _generators_offset: u64,
+        _setup: &Self::ProverPublicSetup<'_>,
```

Comment:
Do you need `_transcript`, `_generators_offset`, and `_setup`?

Reply:
Same comment as above. Setup isn't needed.